src/families/tron/specs.js

```diff
@@ -0,0 +1,152 @@
+import { pickSiblings } from "../../bot/specs";
+import type { AppSpec } from "../../bot/types";
+import deviceActions from "./speculos-deviceActions";
+import { NetworkDown } from "@ledgerhq/errors";
+
+const currency = getCryptoCurrencyById("tron");
+const minimalAmount = parseCurrencyUnit(currency.units[0], "10");
+
+const tron: AppSpec<Transaction> = {
+  name: "Tron",
+  currency,
+  appQuery: {
+    model: "nanoS",
+    appName: "Tron",
+    appVersion: "0.2.x",
+  },
+  mutations: [
+    {
+      name: "move 50% to another account",
+      transaction: ({ account, siblings, bridge }) => {
+        invariant(account.balance.gt(minimalAmount), "balance is too low");
+        let t = bridge.createTransaction(account);
+        const sibling = pickSiblings(siblings);
+        const recipient = sibling.freshAddress;
+        const amount = account.balance.div(2).integerValue();
+        t = bridge.updateTransaction(t, { amount, recipient });
+        return t;
+      },
+      deviceAction: deviceActions.acceptTransaction,
+      test: ({
+        account,
+        accountBeforeTransaction,
+        transaction,
+        optimisticOperation,
+        operation,
+      }) => {
+        // can be generalized!
+        expect(account.balance.toString()).toBe(
+          accountBeforeTransaction.balance.minus(operation.value).toString()
+        );
+      },
+    },
+    {
+      name: "send max to another account",
+      transaction: ({ account, siblings, bridge }) => {
+        invariant(account.balance.gt(minimalAmount), "balance is too low");
+        let t = bridge.createTransaction(account);
+        const sibling = pickSiblings(siblings);
+        const recipient = sibling.freshAddress;
+        t = bridge.updateTransaction(t, { useAllAmount: true, recipient });
+        return t;
+      },
+      deviceAction: deviceActions.acceptTransaction,
+      test: ({
+        account,
+        accountBeforeTransaction,
+        transaction,
+        optimisticOperation,
+        operation,
+      }) => {
+        expect(account.balance.toString()).toBe("0");
+      },
+    },
+    {
+      name: "freeze 25% to bandwidth | energy",
+      transaction: ({ account, bridge }) => {
+        invariant(account.balance.gt(minimalAmount), "balance is too low");
+        let t = bridge.createTransaction(account);
+
+        const amount = account.balance.div(4).integerValue();
+        t = bridge.updateTransaction(t, {
+          mode: "freeze",
+          amount,
+          resource: account.energy.eq(0) ? "ENERGY" : "BANDWIDTH",
+        });
+        return t;
+      },
+      deviceAction: deviceActions.acceptTransaction,
+      test: ({
+        account,
+        accountBeforeTransaction,
+        transaction,
+        optimisticOperation,
+        operation,
+      }) => {
+        const resourceType = transaction.resource;
+        const expectedAmount = BigNumber(transaction.amount)
+          .times(10e6)
```

Comment:
i don't understand why we multiply by 10^6

Reply:
Because the units are not the same between `amount` entered for the transaction and the value of `tronResources.frozen...` value that we use in the test.

Reply:
🤯  this is pretty weird we have this in live-common. transaction.amount is supposed to be in TRON, otherwise we must introduce another field / put in extra.
cc @nukky94  @LFBarreto